crates/rspack_loader_swc/src/utils.rs

```diff
@@ -0,0 +1,33 @@
+pub(crate) struct ParserType {
+  pub(crate) is_typescript: bool,
+  pub(crate) is_jsx: bool,
+}
+// return the parser type of a file based on its extension name
+// tsx -> typescript + jsx
```

Comment:
unfortunately we may also need to consider mjs mts cts cjs, I'm not sure whether there's mtsx ctsx stuff